app/models/components/course/duplicator_component_host.rb

```diff
@@ -0,0 +1,15 @@
+class Duplicator::ModelComponentHost
```

Comment:
so it depends on how the framework is structured. If the framework is structured such that any duplicable object X can duplicate itself, then it is okay for a course to duplicate itself, right?